File: 01-array/0102-27.cpp
/*
 * leetcode 27
 * remove element
 *
 * double-pointer: fast and slow pointers
 * k is the slow pointer and i the fast
 *
 */

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[k++] = nums[i];
            }
        }    
        return k;    
    }
};

File: 01-array/0102-27.go
/*
 * leetcode 27
 * double-pointer: fast and slow
 *
 * _ for something you do not need
 * 
 */

func removeElement(nums []int, val int) int {
    k := 0
    for _, num := range nums {
        if num != val {
            nums[k] = num
            k += 1
        }
    }
    return k    
}
